kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/RequestFactory.java

```diff
@@ -374,9 +398,15 @@ private static void populateDescribeShareGroupOffsetsRequest(ApiMessage apiMessa
         describeShareGroupOffsetsRequest.setGroups(List.of(group));
     }
 
-    private static void populateListConfigResourcesRequest(ApiMessage apiMessage) {
+    private static void populateListConfigResourcesRequest(ApiMessage apiMessage, short apiVersion) {
         final ListConfigResourcesRequestData listConfigResourcesRequestData = (ListConfigResourcesRequestData) apiMessage;
         listConfigResourcesRequestData.setResourceTypes(List.of(ConfigResource.Type.CLIENT_METRICS.id())); // the only type you can use with v0 LIST_CONFIG_RESOURCES
+        if (apiVersion == 0) {
+            ((ListConfigResourcesRequestData) apiMessage).setResourceTypes(List.of());
+        }
+        else {
+            ((ListConfigResourcesRequestData) apiMessage).setResourceTypes(List.of((byte) 16));
```

Comment:
The comment says CLIENT_METRICS is the only type you can use with v0 LIST_CONFIG_RESOURCES, then we proceed to set it is the empty list of the v0 case.  In the > v0 case, we set it back to 16, which is the ordinal value CLIENT_METRICS.   This doesn't make any sense.